endpoints/endpoints.py

```diff
@@ -0,0 +1,1427 @@
+import calendar
+from fabric import Connection
+from invoke import run
+import json
+import logging
+import os
+from pathlib import Path
+import re
+import sys
+import time
+
+TOOLBOX_HOME = os.environ.get('TOOLBOX_HOME')
+if TOOLBOX_HOME is None:
+    print("This script requires libraries that are provided by the toolbox project.")
+    print("Toolbox can be acquired from https://github.com/perftool-incubator/toolbox and")
+    print("then use 'export TOOLBOX_HOME=/path/to/toolbox' so that it can be located.")
+    exit(1)
+else:
+    p = Path(TOOLBOX_HOME) / 'python'
+    if not p.exists() or not p.is_dir():
+        print("ERROR: <TOOLBOX_HOME>/python ('%s') does not exist!" % (p))
+        exit(2)
+    sys.path.append(str(p))
+from toolbox.json import *
+
+ROADBLOCK_HOME = os.environ.get('ROADBLOCK_HOME')
+if ROADBLOCK_HOME is None:
+    print("This script requires libraries that are provided by the roadblock project.")
+    print("Roadblock can be acquired from https://github.com/perftool-incubator/roadblock and")
+    print("then use 'export ROADBLOCK_HOME=/path/to/roadblock so that it can be located.")
+    exit(1)
+else:
+    p = Path(ROADBLOCK_HOME) / 'roadblock.py'
+    if not p.exists() or not p.is_file():
+        print("ERROR: <ROADBLOCK_HOME>/roadblock.py ('%s') does not exist!" % (p))
+        exit(2)
+    sys.path.append(str(Path(ROADBLOCK_HOME)))
+from roadblock import roadblock
+
+roadblock_exits = {
+    "success": 0,
+    "timeout": 3,
+    "abort": 4,
+    "input": 2
+}
```

Comment:
wondering if these numbers need to be updated if they change on roadblock's side

Reply:
They would.  Now, in theory this is the kind of thing that I would never change on the roadblock side in order to maintain a stable "interface".

It does remind me that in the newer model of using roadblock.py as a module that I could potentially refer to these codes from the module that is being imported rather than replicated.  I'll have to look into that.